backend/src/middleware/notifications/mentions-in-grops.spec.ts

```diff
@@ -0,0 +1,845 @@
+    })
+  })
+
+  describe('post in closed group', () => {
+    beforeAll(async () => {
+      jest.clearAllMocks()
+      authenticatedUser = await noMember.toJson()
+      await markAllAsRead()
+      authenticatedUser = await pendingMember.toJson()
+      await markAllAsRead()
+      authenticatedUser = await groupMember.toJson()
+      await markAllAsRead()
+      authenticatedUser = await postAuthor.toJson()
+      await markAllAsRead()
+      await mutate({
+        mutation: createPostMutation,
+        variables: {
+          id: 'closed-post',
+          title: 'This is the post in the closed group',
+          content: `Hey members ${mentionString}! Please read this`,
+          groupId: 'closed-group',
+        },
+      })
+    })
+
+    it('sends NO notification to the no member', async () => {
+      authenticatedUser = await noMember.toJson()
+      await expect(
+        query({
+          query: notificationQuery,
+          variables: {
+            read: false,
+          },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          notifications: [],
+        },
+        errors: undefined,
+      })
+    })
+
+    it('sends NO notification to the pending member', async () => {
+      authenticatedUser = await pendingMember.toJson()
+      await expect(
+        query({
+          query: notificationQuery,
+          variables: {
+            read: false,
+          },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          notifications: [],
+        },
+        errors: undefined,
+      })
+    })
+
+    it('sends a notification to the group member', async () => {
+      authenticatedUser = await groupMember.toJson()
+      await expect(
+        query({
+          query: notificationQuery,
+          variables: {
+            read: false,
+          },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          notifications: [
+            {
+              from: {
+                __typename: 'Post',
+                id: 'closed-post',
+              },
+              read: false,
+              reason: 'post_in_group',
+            },
+            {
+              from: {
+                __typename: 'Post',
+                id: 'closed-post',
+              },
+              read: false,
+              reason: 'mentioned_in_post',
+            },
+          ],
+        },
+        errors: undefined,
+      })
+    })
+
+    it('sends 2 emails, one mention and one post in group', () => {
+      expect(sendMailMock).toHaveBeenCalledTimes(2)
+    })
+  })
+
+  describe('post in hidden group', () => {
+    beforeAll(async () => {
+      jest.clearAllMocks()
+      authenticatedUser = await noMember.toJson()
+      await markAllAsRead()
+      authenticatedUser = await pendingMember.toJson()
+      await markAllAsRead()
+      authenticatedUser = await groupMember.toJson()
+      await markAllAsRead()
+      authenticatedUser = await postAuthor.toJson()
```

Comment:
with an atomic test setup all this will be gone

Reply:
Nope. At least the group member always has a notification, that the role in the group changed